packages/design-system-react-native/src/components/Text/Text.test.tsx

```diff
@@ -0,0 +1,208 @@
+
+        render(<Text fontWeight={weight}>Test</Text>);
+
+        const expectedClassNames = generateClassNames({ fontWeight: weight });
+        expect(twMock).toHaveBeenCalledWith(expectedClassNames);
+      });
+    });
+  });
+
+  describe('Font Style', () => {
+    Object.values(FontStyle).forEach((style) => {
+      it(`applies font style ${style} correctly`, () => {
+        const twMock = jest.fn().mockReturnValue({});
+        (useTailwind as jest.Mock).mockReturnValue(twMock);
+
+        render(<Text fontStyle={style}>Test</Text>);
+
+        const expectedClassNames = generateClassNames({ fontStyle: style });
+        expect(twMock).toHaveBeenCalledWith(expectedClassNames);
+      });
+    });
+  });
+
+  it('renders children correctly', () => {
+    const { getByText } = render(
+      <Text variant={TextVariant.BodyMd}>Hello, World!</Text>,
+    );
+    const textElement = getByText('Hello, World!');
+    expect(textElement).not.toBeNull();
+  });
+
+  it('renders with default color when no color prop is provided', () => {
+    const twMock = jest.fn().mockReturnValue({});
+    (useTailwind as jest.Mock).mockReturnValue(twMock);
+
+    render(<Text variant={TextVariant.BodyMd}>Default Color Text</Text>);
+
+    const expectedClassNames = generateClassNames({
+      variant: TextVariant.BodyMd,
+    });
+    expect(twMock).toHaveBeenCalledWith(expectedClassNames);
+  });
+
+  it('renders with default variant (BodyMd) when no variant is provided', () => {
+    const twMock = jest.fn().mockReturnValue({});
+    (useTailwind as jest.Mock).mockReturnValue(twMock);
+
+    render(<Text>Default Variant Text</Text>);
+
+    const expectedClassNames = generateClassNames({});
+    expect(twMock).toHaveBeenCalledWith(expectedClassNames);
+  });
+
+  it('applies accessibilityRole="text"', () => {
+    const { getByText } = render(<Text>Accessible Text</Text>);
+    const textElement = getByText('Accessible Text');
+    expect(textElement.props.accessibilityRole).toBe('text');
+  });
+
+  it('passes additional props to RNText', () => {
+    const onPressMock = jest.fn();
+    const { getByText } = render(
+      <Text onPress={onPressMock}>Pressable Text</Text>,
+    );
+    const textElement = getByText('Pressable Text');
+    expect(textElement.props.onPress).toBe(onPressMock);
+  });
+
+  it('combines style prop with generated styles', () => {
+    const customStyle = { margin: 10 };
+    const { getByText } = render(<Text style={customStyle}>Styled Text</Text>);
+    const textElement = getByText('Styled Text');
+    expect(textElement.props.style).toContain(customStyle);
+  });
+
+  it('applies custom class names from twClassNames', () => {
+    const twMock = jest.fn().mockReturnValue({});
+    (useTailwind as jest.Mock).mockReturnValue(twMock);
+
+    render(<Text twClassNames="custom-class">Custom Class Text</Text>);
+
+    expect(twMock).toHaveBeenCalledWith(
+      expect.stringContaining('custom-class'),
+    );
+  });
```

Comment:
Oh if this is the case maybe we should just use the `className` prop for consistency across platforms?

Reply:
I still want to discourage non-tw classnames